src/common/ItemIcon.tsx

```diff
@@ -7,16 +7,15 @@ import Icon from './Icon';
 type Props = {
   id: number;
   noLink?: boolean;
-  details?: Record<string, any>;
```

Comment:
It looks like the `details`-prop is used in [ItemLink](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/271b3d751735258c15dce435877cf4c2f1dda6ce/src/common/ItemLink.js#L39) to override the quality of an item (because the fallback is using the quality from `ITEMS`)

Some Nyalotha items use it without prividing `quality`, so we *should* keep it, although `ItemLink` is still using iLvl 370 to determine if blue or purple (so it isn't reliable right now anyways).

Reply:
Actually, no `<ItemIcon />` isn't using `details` right now, but I think we should keep it if someone needs to overwrite it in the future.